ports/savingprincess/Saving Princess.sh

```diff
@@ -0,0 +1,157 @@
+#!/bin/bash
+
+XDG_DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
+
+if [ -d "/opt/system/Tools/PortMaster/" ]; then
+  controlfolder="/opt/system/Tools/PortMaster"
+elif [ -d "/opt/tools/PortMaster/" ]; then
+  controlfolder="/opt/tools/PortMaster"
+elif [ -d "$XDG_DATA_HOME/PortMaster/" ]; then
+  controlfolder="$XDG_DATA_HOME/PortMaster"
+else
+  controlfolder="/roms/ports/PortMaster"
+fi
+
+source $controlfolder/control.txt
+source $controlfolder/device_info.txt
+export PORT_32BIT="Y"
+
+get_controls
+[ -f "${controlfolder}/mod_${CFW_NAME}.txt" ] && source "${controlfolder}/mod_${CFW_NAME}.txt"
+
+$ESUDO chmod 666 /dev/tty0
+
+GAMEDIR="/$directory/ports/savingprincess"
+
+export LD_LIBRARY_PATH="/usr/lib32:$GAMEDIR/libs:$LD_LIBRARY_PATH"
+export GMLOADER_DEPTH_DISABLE=1
+export GMLOADER_SAVEDIR="$GAMEDIR/gamedata/"
+export GMLOADER_PLATFORM="os_windows"
+export TOOLDIR="$GAMEDIR/tools"
+export PATH=$PATH:$GAMEDIR/tools
+
+# We log the execution of the script into log.txt
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+
+# Permissions
+$ESUDO chmod 666 /dev/uinput
+$ESUDO chmod +x "$GAMEDIR/gmloader"
+$ESUDO chmod 777 "$TOOLDIR/gmKtool.py"
+
+cd $GAMEDIR
+
+#Patch Game
+
+if [ -f "$GAMEDIR/gamedata/Saving Princess.exe" ]; then
```

Comment:
Looks like this will run every launch if the file exists?

It doesnt appear to delete them after extraction.

Reply:
# Check if the Saving Princess - DEMO.exe file exists
        if [ -f "$GAMEDIR/gamedata/Saving Princess v0_8.exe" ]; then
            # Delete the redundant .exe files
            rm -f "$GAMEDIR/gamedata/Saving Princess v0_8.exe"
            rm -f "$GAMEDIR/gamedata/Saving Princess.exe"
	    rm -f "$GAMEDIR/gamedata/D3DX9_43.dll"
        fi
        
Apologies the comment is wrong will edit this after I am back

Reply:
Nah its fine, i must've not seen it pre-coffee. 😓